tests/check/check_complexity4.go

```diff
@@ -0,0 +1,300 @@
+				Expectation: true,
+			},
+			{
+				Name:        "no_path_to_employee",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:ps1", Relation: "userset_or_compute_complex3", User: "employee:invalid2"},
+				Expectation: false,
+			},
+		},
+	},
+	{
+		Name: "complexity4_ttu_and_nested_complex3",
+		// not possible to reach an employee
+		// 2 possible paths to reach a user
+		Tuples: []*openfgav1.TupleKey{
+			{Object: "complexity4:pe1", Relation: "parent", User: "complexity3:pe2"},
+			{Object: "complexity3:pe2", Relation: "and_nested_complex3", User: "ttus:pe3#and_ttu"},
+			{Object: "directs-user:pe4", Relation: "direct", User: "user:valid"},
+			{Object: "complexity3:pe2", Relation: "userset_parent", User: "usersets-user:pe3"},
+			{Object: "usersets-user:pe3", Relation: "ttu_direct_userset", User: "ttus:pe3#direct_pa_direct_ch"},
+			{Object: "ttus:pe3", Relation: "mult_parent_types", User: "directs-user:pe4"},
+			// path 1 in the union
+			{Object: "ttus:pe3", Relation: "direct_parent", User: "directs-user:pe4"},
+			// path 2 in the union that cannot lead to truthy because we are missing tuples on other parts of the graph
+			{Object: "directs-user:pe4", Relation: "direct_cond", User: "user:validwithcond", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+			// path 3 for employee cannot lead to truthy because we are missing tuples on other parts of the graph
+			{Object: "ttus:pe3", Relation: "mult_parent_types", User: "directs-employee:pe4"},
+			{Object: "directs-employee:pe4", Relation: "direct", User: "employee:invalid"},
+		},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "path_to_user_1",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:pe1", Relation: "ttu_and_nested_complex3", User: "user:valid"},
+				Expectation: true,
+			},
+			{
+				Name:        "no_path_to_user_with_cond_truthy",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:pe1", Relation: "ttu_and_nested_complex3", User: "user:validwithcond"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: false,
+			},
+			{
+				Name:        "no_path_to_employee",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:pe1", Relation: "ttu_and_nested_complex3", User: "employee:invalid"},
+				Expectation: false,
+			},
+		},
+	},
+	{
+		Name: "complexity4_or_complex4",
+		// reuse tuples from the previous two tests
+		Tuples: []*openfgav1.TupleKey{},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "path_to_user_through_ttu_and_nested_complex3",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:pe1", Relation: "or_complex4", User: "user:valid"},
+				Expectation: true,
+			},
+			{
+				Name:        "path_to_user_through_userset_or_compute_complex3",
+				Tuple:       &openfgav1.TupleKey{Object: "complexity4:ps1", Relation: "or_complex4", User: "user:valid"},
```

Comment:
Can you verify whether we want `ps1` or `pe1`?  (also apply to the next 2 lines)

Reply:
Sorry I don't understand, `pe1` is in the assertion above